internal/runner/rspec.go

```diff
@@ -112,3 +113,71 @@ func (r Rspec) commandNameAndArgs(testCases []string) (string, []string, error)
 	words = slices.Replace(words, idx, idx+1, testCases...)
 	return words[0], words[1:], nil
 }
+
+// RspecExample represents a single test example in an Rspec report.
+type RspecExample struct {
+	Id              string  `json:"id"`
+	Description     string  `json:"description"`
+	FullDescription string  `json:"full_description"`
+	Status          string  `json:"status"`
+	FilePath        string  `json:"file_path"`
+	LineNumber      int     `json:"line_number"`
+	RunTime         float64 `json:"run_time"`
+}
+
+// RspecReport is the structure for Rspec JSON report.
+type RspecReport struct {
+	Version  string         `json:"version"`
+	Seed     int            `json:"seed"`
+	Examples []RspecExample `json:"examples"`
+}
+
+// GetExamples returns an array of test examples within the given files.
+func (r Rspec) GetExamples(files []string) ([]plan.TestCase, error) {
+	// Create a temporary file to store the JSON output of the rspec dry run.
+	// We cannot simply read the dry run output from stdout because
+	// users may have custom formatters that do not output JSON.
+	f, err := os.CreateTemp("", "dry-run-*.json")
+	if err != nil {
+		return []plan.TestCase{}, fmt.Errorf("failed to create temporary file for rspec dry run: %v", err)
+	}
+	defer f.Close()
+	defer os.Remove(f.Name())
+
+	cmdName, cmdArgs, err := r.commandNameAndArgs(files)
+	if err != nil {
+		return nil, err
+	}
+
+	cmdArgs = append(cmdArgs, "--dry-run", "--format", "json", "--out", f.Name())
+
+	output, err := exec.Command(cmdName, cmdArgs...).CombinedOutput()
+
+	outputStr := string(output)
+	if err != nil {
+		return []plan.TestCase{}, fmt.Errorf("failed to run rspec dry run: %v", outputStr)
```

Comment:
Do we want to wrap the `err` or `outputStr` with the message? 
I was thinking:
`fmt.Errorf("failed to run rspec dry run: %v", err)`

Do we need to move error handling to line 155?

Reply:
I did that initially, but the message from output is more descriptive.

```
fmt.Errorf("failed to run rspec dry run: %v", err)
> failed to run rspec dry run: exit status 10
```
vs
```
fmt.Errorf("failed to run rspec dry run: %v", outputStr)
> failed to run rspec dry run: Could not locate Gemfile or .bundle/ directory
```

Reply:
Also, `CombinedOutput` will return output from both `stdout` and `stderr`.

Reply:
Yeah, I think it gives more meaningful message if we wrap `outputStr`

Just curious, when dry run fails, does the program exit with `10` or `1`? Do we want do document the exit code in the README?